### PR TITLE
PlayStation Vita support

### PR DESCRIPTION
**Pull request by fgsfdsfgs**
Adds PSVita support to both waf and cmake scripts. Complementary to FWGS/xash3d-fwgs#1251.
Building the libraries for `psvita` requires the same dependencies as building the engine for it.

**Building with waf:**
```
./waf configure -T release --psvita
./waf build
```

**Building with CMake:**
```
mkdir build && cd build
cmake -G"Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="$VITASDK/share/vita.toolchain.cmake" -DCMAKE_PROJECT_HLSDK-PORTABLE_INCLUDE="$VITASDK/share/vrtld_shim.cmake" ..
make -j
```

### CODE REVIEW

File: scripts/waifulib/xcompile.py [hunk 1]
```diff
@@ -434,6 +434,75 @@ def ldflags(self):
...
+	def cflags(self, cxx = False):
+		cflags = []
+		# arch flags
+		cflags += ['-D__vita__', '-mtune=cortex-a9', '-mfpu=neon']
+		# necessary linker flags
+		cflags += ['-Wl,-q', '-Wl,-z,nocopyreloc']
+		# this optimization is broken in vitasdk
+		cflags += ['-fno-optimize-sibling-calls']
+		# disable some ARM bullshit
+		cflags += ['-fsigned-char', '-fno-short-enums', '-Wno-attributes']
```
Comment (a1batross) [inline scripts/waifulib/xcompile.py:483]: We should be able to build without `-fsigned-char` now.

But `-fno-short-enums` should be necessary.

File: scripts/waifulib/xcompile.py [hunk 2]
```diff
@@ -434,6 +434,75 @@ def ldflags(self):
...
+		# base include dir
+		cflags += ['-isystem %s/arm-vita-eabi/include' % self.vitasdk_dir]
+		# SDL include dir
+		cflags += ['-I%s/arm-vita-eabi/include/SDL2' % self.vitasdk_dir]
+		# the game wants GNU extensions
+		if cxx:
+			cflags += ['-std=gnu++17', '-D_GNU_SOURCE']
```
Comment (a1batross) [inline scripts/waifulib/xcompile.py:490]: Don't we enable _GNU_SOURCE only when we actually use it?

File: scripts/waifulib/xcompile.py [hunk 3]
```diff
@@ -434,6 +434,75 @@ def ldflags(self):
...
+		# necessary linker flags
+		cflags += ['-Wl,-q', '-Wl,-z,nocopyreloc']
+		# this optimization is broken in vitasdk
+		cflags += ['-fno-optimize-sibling-calls']
+		# disable some ARM bullshit
+		cflags += ['-fsigned-char', '-fno-short-enums', '-Wno-attributes']
+		# base include dir
+		cflags += ['-isystem %s/arm-vita-eabi/include' % self.vitasdk_dir]
+		# SDL include dir
+		cflags += ['-I%s/arm-vita-eabi/include/SDL2' % self.vitasdk_dir]
```
Comment (a1batross) [inline scripts/waifulib/xcompile.py:487]: SDL2 shouldn't be added globally here and instead be in uselib.

But this is a minor issue, I wouldn't change it right now but refactor it in engine later. Maybe I would do that myself.

File: scripts/waifulib/xcompile.py (hunk 1 above)
Comment (fgsfdsfgs) [reply scripts/waifulib/xcompile.py]: `-fsigned-char` is actually the default in vitasdk gcc now, so removed.

File: scripts/waifulib/xcompile.py (hunk 2 above)
Comment (fgsfdsfgs) [reply scripts/waifulib/xcompile.py]: Yeah, I don't remember why I put that in. Removed.

**Comment by nekonomicon** [comment]
@fgsfdsfgs please add build instructions to readme.md.

**Comment by a1batross** [comment]
@nekonomicon do you really wanna pile up all build instructions in one single readme md file?

**Comment by nekonomicon** [comment]
@a1batross [issue for any suggestions](https://github.com/FWGS/hlsdk-portable/issues/263)

**Comment by fgsfdsfgs** [comment]
> @fgsfdsfgs please add build instructions to readme.md.

Done.

**Pull request merged**